cohortextractor/query_engines/base_sql.py

```diff
@@ -64,17 +67,16 @@ def get_primary_table(query):
 
 class BaseSQLQueryEngine(BaseQueryEngine):
 
-    sqlalchemy_dialect = NotImplemented
+    sqlalchemy_dialect: Union[type[Dialect], ModuleType]
```

Comment:
Gross! I've looked into why this type is so weird and don't understand what's going on. Nothing for you to do here, but this is a good example of why clarifying the types is helpful.